build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -73,8 +85,22 @@ async function generateReleaseReport(
 	const simpleReportOutput = toReportKind(fullReleaseReport, "simple");
 
 	await Promise.all([
-		writeReport(outDir, caretReportOutput as ReleaseReport, "manifest", version, log),
-		writeReport(outDir, simpleReportOutput as ReleaseReport, "simpleManifest", version, log),
+		writeReport(
+			outDir,
+			caretReportOutput as ReleaseReport,
+			"manifest",
+			version,
+			branchName,
+			log,
+		),
+		writeReport(
+			outDir,
+			simpleReportOutput as ReleaseReport,
+			"simpleManifest",
+			version,
+			branchName,
+			log,
+		),
```

Comment:
It feels to me like `writeReport()` is becoming too complicated for its own good. What do you think of refactoring a bit so that the build-number manifest gets generated here unconditionally (we want it for any branch after all), and have a single `if (<branch is main>) { <generate manifest by date> }`. This function can compute the filenames and directly call `fs.writeFile()`. Also saves a tiny bit of compute time because we won't have to compute and format the date twice when we only use it in one case (super minor benefit, but why not 😄).

Reply:
that's a good idea. updated the PR :)